k8s-genie/scripts/provision.sh

```diff
@@ -0,0 +1,168 @@
+
+# Disable swap
+swapoff -a
+sed -i '/ swap / s/^/#/' /etc/fstab
+
+# Disable spectre and meltdown patches
+sed -i 's/quiet"/quiet spectre_v2=off nopti hugepagesz=2M hugepages=64"/' /etc/default/grub
+grub2-mkconfig -o /boot/grub2/grub.cfg
+
+systemctl stop firewalld NetworkManager || :
+systemctl disable firewalld NetworkManager || :
+# Make sure the firewall is never enabled again
+# Enabling the firewall destroys the iptable rules
+yum -y remove NetworkManager firewalld
+
+# Required for iscsi demo to work.
+yum -y install iscsi-initiator-utils
+
+cat <<EOF >/etc/yum.repos.d/kubernetes.repo
+[kubernetes]
+name=Kubernetes
+baseurl=http://yum.kubernetes.io/repos/kubernetes-el7-x86_64
+enabled=1
+gpgcheck=1
+repo_gpgcheck=1
+gpgkey=https://packages.cloud.google.com/yum/doc/yum-key.gpg
+       https://packages.cloud.google.com/yum/doc/rpm-package-key.gpg
+EOF
+yum install -y docker
+
+# Log to json files instead of journald
+sed -i 's/--log-driver=journald //g' /etc/sysconfig/docker
+echo '{ "insecure-registries" : ["registry:5000"] }' > /etc/docker/daemon.json
+
+# Enable the permanent logging
+# Required by the fluentd journald plugin
+# The default settings in recent distribution for systemd is set to auto,
+# when on auto journal is permament when /var/log/journal exists
+mkdir -p /var/log/journal
+
+# Omit pgp checks until https://github.com/kubernetes/kubeadm/issues/643 is resolved.
+yum install --nogpgcheck -y \
+    kubeadm-${version} \
+    kubelet-${version} \
+    kubectl-${version} \
+    kubernetes-cni
+
+# Latest docker on CentOS uses systemd for cgroup management
+# kubeadm 1.11 uses a new config method for the kubelet
+if [[ $version =~ \.([0-9]+) ]] && [[ ${BASH_REMATCH[1]} -ge "11" ]]; then
+    # TODO use config file! this is deprecated
+    cat <<EOT >/etc/sysconfig/kubelet
+KUBELET_EXTRA_ARGS=--cgroup-driver=systemd --runtime-cgroups=/systemd/system.slice --kubelet-cgroups=/systemd/system.slice --feature-gates=BlockVolume=true
+EOT
+else
+    cat <<EOT >>/etc/systemd/system/kubelet.service.d/09-kubeadm.conf
+[Service]
+Environment="KUBELET_EXTRA_ARGS=--cgroup-driver=systemd --runtime-cgroups=/systemd/system.slice --kubelet-cgroups=/systemd/system.slice --feature-gates=BlockVolume=true"
+EOT
+fi
+
+systemctl daemon-reload
+
+systemctl enable docker && systemctl start docker
+systemctl enable kubelet && systemctl start kubelet
+
+# Needed for kubernetes service routing and dns
+# https://github.com/kubernetes/kubernetes/issues/33798#issuecomment-250962627
+modprobe bridge
+cat <<EOF >  /etc/sysctl.d/k8s.conf
+net.bridge.bridge-nf-call-ip6tables = 1
+net.bridge.bridge-nf-call-iptables = 1
+EOF
+sysctl --system
+
+kubeadm init --pod-network-cidr=10.244.0.0/16 --kubernetes-version v${version} --token abcdef.1234567890123456
+
+# install genie
+curl https://raw.githubusercontent.com/Huawei-PaaS/CNI-Genie/master/conf/1.8/genie-plugin.yaml --output /etc/kubernetes/genie.yml
```

Comment:
Can you please download the file into the manifest folder and use it from there?